retriever/lib/repository.py

```diff
@@ -12,80 +12,96 @@
 from retriever import REPOSITORY, SCRIPT_WRITE_PATH, HOME_DIR
 from retriever.lib.models import file_exists
 
-global abort, executable_name
-abort = False
-executable_name = "retriever"
 
-
-def download_from_repository(filepath, newpath, repo=REPOSITORY):
+def _download_from_repository(filepath, newpath, repo=REPOSITORY):
     """Downloads the latest version of a file from the repository."""
     try:
-        filename = filepath.split('/')[-1]
         urllib.request.urlretrieve(repo + filepath, newpath)
     except:
         raise
         pass
 
 
-def check_for_updates():
+def update_scripts(script_names=None):
     """Check for updates to scripts.
 
     This updates the HOME_DIR scripts directory with the latest script versions
+    when no script_names are specified.
+    Updates specific scripts when `script_names` passed as argument.
     """
     try:
-        # open version.txt for current release branch and get script versions
-        version_file = urllib.request.urlopen(REPOSITORY + "version.txt")
-        version_file.readline()
+        if script_names is None:
+            version_file = urllib.request.urlopen(REPOSITORY + "version.txt")
+            version_file.readline()
 
-        # read scripts from the repository and the checksums from the version.txt
-        scripts = []
-        print("Downloading scripts...")
-        for line in version_file:
-            scripts.append(line.decode().strip('\n').split(','))
+            scripts = []
+            print("Downloading scripts...")
+            for line in version_file:
+                scripts.append(line.decode().strip('\n').split(','))
+        else:
+            scripts = script_names
```

Comment:
This approach works well, but I'll mention one alternative in case you like it better. If we name the optional argument `scripts` instead of `script_names` and we set it's default value to `[]` instead of `None` then we could use something like this:

```
if not scripts:
    version_file = urllib.request.urlopen(REPOSITORY + "version.txt")
    version_file.readline()
    print("Downloading scripts...")
    for line in version_file:
        scripts.append(line.decode().strip('\n').split(','))
````

This lets us drop a couple of lines and also helps it read a little more directly as "If there are no scripts provided, use the list in the version file". I don't think it's a big improvement, but just something for you to consider in case you thing it reads better.